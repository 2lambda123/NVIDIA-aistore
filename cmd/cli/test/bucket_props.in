ais bucket create ais://$BUCKET_1
ais bucket props ais://$BUCKET_1 'mirror.enabled=true' 'mirror.copies=2'

ais bucket props ais://$BUCKET_1 'unknown.field' // FAIL "invalid property "unknown.field""
ais bucket props ais://$BUCKET_1 'unknown.field=true' // FAIL "unknown property "unknown.field""
ais bucket props ais://$BUCKET_1 'provider=ais' // FAIL "property "provider" is readonly"

ais show bucket ais://$BUCKET_1
ais bucket props --reset ais://$BUCKET_1

ais bucket props ais://$BUCKET_1 '{"versioning": {"enabled": true, "validate_warm_get": true}}'
ais show bucket ais://$BUCKET_1
ais show bucket ais://$BUCKET_1 lru -v

ais bucket props ais://$BUCKET_1 '{"versioning": {"enabled": true, "validate_warm_get": true}}'
