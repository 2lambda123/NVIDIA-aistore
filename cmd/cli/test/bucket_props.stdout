"$BUCKET_1" bucket created
Bucket props successfully updated
PROPERTY	 VALUE
access		 GET,HEAD-OBJECT,PUT,APPEND,COLD-GET,DELETE-OBJECT,RENAME-OBJECT,PROMOTE,HEAD-BUCKET,LIST-OBJECTS,RENAME-BUCKET,PATCH,MAKE-NCOPIES,SYNC-BUCKET,DELETE-BUCKET
checksum	 Type: xxhash | Validate: ColdGET
^created.*$
ec		 Disabled
lru		 Watermarks: 75%/90% | Do not evict time: 120m | OOS: 95%
mirror		 2 copies
provider	 ais
versioning	 Enabled | Validate on WarmGET: no
Bucket props successfully reset
Bucket props successfully updated
PROPERTY	 VALUE
access		 GET,HEAD-OBJECT,PUT,APPEND,COLD-GET,DELETE-OBJECT,RENAME-OBJECT,PROMOTE,HEAD-BUCKET,LIST-OBJECTS,RENAME-BUCKET,PATCH,MAKE-NCOPIES,SYNC-BUCKET,DELETE-BUCKET
checksum	 Type: xxhash | Validate: ColdGET
^created.*$
ec		 Disabled
lru		 Watermarks: 75%/90% | Do not evict time: 120m | OOS: 95%
mirror		 Disabled
provider	 ais
versioning	 Enabled | Validate on WarmGET: yes
PROPERTY		 VALUE
lru.capacity_upd_time	 10m
lru.dont_evict_time	 120m
lru.enabled		 true
lru.highwm		 90
lru.lowwm		 75
lru.out_of_space	 95
